### PR TITLE
pnpm setup

### PR DESCRIPTION
**Pull request by estrattonbailey**
Replaces `yarn` with `pnpm` and fixes a few dozen type errors, dependency mismatches, and missing dependencies.

To test, checkout this branch and run through all commands:
```
pnpm i

pnpm build

pnpm verify

pnpm format

pnpm test
```

When publishing, the process should be essentially the same and we should be able to use any flavor of `publish` we want e.g. `npm` `yarn` or `pnpm`. My plan is to add Changesets to automate this in a future PR, at which point we won't need to worry about manually publishing anymore.

Todo:
- [x] test publishing with `publishConfig`
- [x] test Docker builds
- [x] test run Docker image

### CODE REVIEW

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+enable-pre-post-scripts = true
```
Comment (estrattonbailey) [inline .npmrc:1]: pnpm doesn't enable these by default

File: packages/api/package.json [hunk 2]
```diff
@@ -2,15 +2,16 @@
...
-    "codegen": "yarn docgen && node ./scripts/generate-code.mjs && lex gen-api ./src/client ../../lexicons/com/atproto/*/* ../../lexicons/app/bsky/*/*",
+    "codegen": "pnpm docgen && node ./scripts/generate-code.mjs && lex gen-api ./src/client ../../lexicons/com/atproto/*/* ../../lexicons/app/bsky/*/*",
     "docgen": "node ./scripts/generate-docs.mjs",
     "build": "node ./build.js",
     "postbuild": "tsc --build tsconfig.build.json",
-    "update-main-to-dist": "node ./update-pkg.js --update-main-to-dist",
-    "update-main-to-src": "node ./update-pkg.js --update-main-to-src",
-    "prepublish": "npm run update-main-to-dist",
-    "postpublish": "npm run update-main-to-src",
+    "update-main-to-dist": "node ../../update-main-to-dist.js packages/api",
```
Comment (estrattonbailey) [inline packages/api/package.json:14]: We still need this functionality unfortunately, but only for the runtime. I pulled the script up to the root for reuse.

File: packages/api/update-pkg.js
Comment (estrattonbailey) [inline packages/api/update-pkg.js:1]: This script has been pulled up to the root of the repo for reuse

File: packages/bsky/src/image/logger.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import { subsystemLogger } from '@atproto/common'
 
-export const logger = subsystemLogger('bsky:image')
+export const logger: ReturnType<typeof subsystemLogger> =
+  subsystemLogger('bsky:image')
```
Comment (estrattonbailey) [inline packages/bsky/src/image/logger.ts:4]: Fixes a kinda weird typescript bug

File: packages/repo/src/types.ts [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import { z } from 'zod'
-import { schema as common, def as commonDef } from '@atproto/common'
+import { def as commonDef } from '@atproto/common-web'
+import { schema as common } from '@atproto/common'
```
Comment (estrattonbailey) [inline packages/repo/src/types.ts:3]: Fixed another weird typescript error by doing this

File: update-main-to-dist.js
Comment (estrattonbailey) [inline update-main-to-dist.js:1]: The new script based on the old ones

**Review by dholms — APPROVED** [review]
just gave it a go in stagin & it worked like a charm 👌

**Pull request merged**